zipline/finance/slippage.py

```diff
@@ -63,8 +63,20 @@ def simulate(self, bar_data, asset, orders_for_asset):
             if not order.triggered:
                 continue
 
+            txn = None
+
             try:
-                txn = self.process_order(bar_data, price, volume, order)
+                execution_price, execution_volume = \
+                    self.process_order(data, order)
+
+                if execution_price is not None:
+                    txn = create_transaction(
+                        order,
+                        data.current_dt,
```

Comment:
Should this function just take data (or have data baked in) so that the real `current_dt` is always enforced?